module-bsp/board/rt1051/puretx/bsp/keyboard/keyboard.cpp

```diff
@@ -210,15 +259,15 @@ namespace bsp::keyboard
             uint8_t retval = 0;
             // Read how many key events has been registered
             i2cAddr.subAddress = REG_KEY_LCK_EC;
-            i2c->Read(i2cAddr, reinterpret_cast<std::uint8_t *>(&retval), 1);
+            read(i2cAddr, reinterpret_cast<std::uint8_t *>(&retval), 1);
 
             std::uint8_t eventsCount = retval & keyCountRegisterMask;
             for (std::uint8_t i = 0; i < eventsCount; i++) {
                 std::uint8_t key      = 0;
                 std::uint8_t rel_pres = 0;
 
                 i2cAddr.subAddress = REG_KEY_EVENT_A;
-                i2c->Read(i2cAddr, reinterpret_cast<std::uint8_t *>(&retval), 1);
+                read(i2cAddr, reinterpret_cast<std::uint8_t *>(&retval), 1);
 
```

Comment:
here I think we should check if the command succeeded, although as we don't have any mechanism to restart the I2C with success (as @Lefucjusz said) it's not critical for now

Reply:
We don't know why the keyboard fails sometimes. The `read` function is trying to retransmit if something went wrong. If it doesn't help we just log this event. The resetting chip or other actions will be considered in the future when we find an exact problem.

Reply:
alright